src/ngx_http_lua_kong_module.c

```diff
@@ -471,4 +531,91 @@ ngx_http_lua_kong_ffi_set_upstream_client_cert_and_key(ngx_http_request_t *r,
     return NGX_ERROR;
 }
 
+int
+ngx_http_lua_kong_ffi_set_upstream_ssl_trusted_store(ngx_http_request_t *r,
+    void *_store)
+{
+    X509_STORE                  *store = _store;
+    ngx_http_lua_kong_ctx_t     *ctx;
+
+    if (store == NULL) {
+        return NGX_ERROR;
+    }
+
+    ctx = ngx_http_lua_kong_get_module_ctx(r);
+    if (ctx == NULL) {
+        return NGX_ERROR;
+    } else if (ctx->upstream_client_certificate_chain != NULL) {
+        ngx_http_lua_kong_cleanup_trusted_store(ctx);
+
+        ctx->upstream_trusted_store = NULL;
+    }
+
+    if (X509_STORE_up_ref(store) == 0) {
+        goto failed;
+    }
+
+    ctx->upstream_trusted_store = store;
+
+    return NGX_OK;
+
+failed:
+
+    ERR_clear_error();
+
+    return NGX_ERROR;
+}
+
+int
+ngx_http_lua_kong_ffi_set_upstream_ssl_verify(ngx_http_request_t *r,
+    int verify, int depth)
+{
+    ngx_http_lua_kong_ctx_t     *ctx;
+
+    ctx = ngx_http_lua_kong_get_module_ctx(r);
+    if (ctx == NULL) {
+        return NGX_ERROR;
+    }
+
+    ctx->upstream_ssl_verify = verify ?
+            NGX_HTTP_LUA_KONG_SSL_VERIFY_ON : NGX_HTTP_LUA_KONG_SSL_VERIFY_OFF;
+
+
+    return NGX_OK;
+}
+
+int
+ngx_http_lua_kong_ffi_set_upstream_ssl_verify_depth(ngx_http_request_t *r,
+    int depth)
+{
+    ngx_http_lua_kong_ctx_t     *ctx;
+
+    ctx = ngx_http_lua_kong_get_module_ctx(r);
+    if (ctx == NULL) {
+        return NGX_ERROR;
+    }
+
+    /* ssl_verify_depth can be set to 0 as well (self-signed only), add a marker
+    * so later we know this flag is modified */
+    ctx->upstream_ssl_verify_depth = depth << 1 | 0x1;
```

Comment:
Same here, we can use `NGX_CONF_UNSET_UINT` http://lxr.nginx.org/source/src/core/ngx_conf_file.h#0057 instead.